docs/DeveloperGuide.md

```diff
@@ -1,33 +1,244 @@
+
+## Setting up the project in your computer
+**Prerequisites:**
+* JDK 11
+* IntelliJ IDE
+
+<div markdown="span" class="alert alert-warning">
+
+⚠ **Caution:**
+
+Follow the steps in the following guide precisely. Things will not work out if you deviate in some steps.
+</div>
+
+First, **fork** this repo, and **clone** the fork into your computer.
+
+If you plan to use Intellij IDEA (highly recommended):
+1. **Configure the JDK**: Follow the guide [_[se-edu/guides] IDEA: Configuring the JDK_](https://se-education.org/guides/tutorials/intellijJdk.html) to to ensure Intellij is configured to use **JDK 11**.
+1. **Import the project as a Gradle project**: Follow the guide [_[se-edu/guides] IDEA: Importing a Gradle project_](https://se-education.org/guides/tutorials/intellijImportGradleProject.html) to import the project into IDEA.<br>
+  :exclamation: Note: Importing a Gradle project is slightly different from importing a normal Java project.
+1. **Verify the setup**:
+   1. Run the `java -jar duke.jar` and try a few commands.
+   2. [Run the tests](Testing.md) to ensure they all pass. 
+//Note: should ask team W11-02 the permission for the above part.
+
+## System Overview
+
+The following sections describe the high level overview of our application, Fluffle.
+
+### Technological view
+The Integrated Development Environment, IntelliJ, is used to develop our program. The program is written in Java, 
+and uses Gradle for building and testing purposes. Our source code is mostly original, with some functions imported 
+from the java.util package. The remaining packages and classes which form the structure of our program 
+are independently developed.
+
+## Architecture
+![UML Fluffle class diagram](graphics/diagrams/classDiagram_Fluffle.png)
+<center><i><b>Figure 1: Overview of Fluffle UML diagram</b></i></center>
+
+Fluffle Documentation: This document contains all packages and classes that are used in developing Fluffle. 
+
+### Project overview
+Fluffle is built using IntelliJ and all concepts for the user interfaces, as well as the backend data management of the 
+application, was created by our team. Due to the restrictions of the project, the main file format used for storage is 
+.txt. We opted to save the data in the text files in a user readable format as opposed to the comma separated format 
+as it is easier for users to directly refer to and edit their saved files.
+
+## Writings class family
+### Constitution (member classes)
+WritingList: Represent the objects which are particular lists of Writings to be used in the application.
+Writings: Represent the objects of the writings, created from user’s input and stored in a database as text. 
+This Writings class is also the parent of 2 subclasses which are Poem and Essay.
+Poem: Represents the Writings objects which have type is Poem.
+Essay: Represents the Writings objects which have type is Essay.
+User: Represents the Users registered to the System
```

Comment:
These definitions are spaced out on the file here but on the github pages link or the actual developerguide.md on github they are not spaced out properly and are kind of hard to read. Maybe you could space them out with each on a single line?